orderapi/src/main/java/fi/hel/verkkokauppa/order/service/accounting/AccountingSlipService.java

```diff
@@ -0,0 +1,201 @@
+        );
+    }
+
+    public List<AccountingSlipRow> getAccountingSlipRows(String accountingSlipId) {
+        List<AccountingSlipRow> rows = accountingSlipRowRepository.findByAccountingSlipId(accountingSlipId);
+
+        if (!rows.isEmpty()) {
+            return rows;
+        }
+
+        throw new CommonApiException(
+                HttpStatus.NOT_FOUND,
+                new Error("accounting-slip-rows-not-found-from-backend", "accounting slip rows with accountingSlipId [" + accountingSlipId + "] not found from backend")
+        );
+    }
+
+    public AccountingSlipDto transformAccountingSlipWithRowsToDto(AccountingSlip accounting, List<AccountingSlipRow> accountingSlipRows) {
+        AccountingSlipDto accountingSlipDto = new AccountingSlipTransformer().transformToDto(accounting);
+        List<AccountingSlipRowDto> accountingSlipRowDtos = new ArrayList<>();
+        accountingSlipRows.forEach(row -> accountingSlipRowDtos.add(new AccountingSlipRowTransformer().transformToDto(row)));
+        accountingSlipDto.setRows(accountingSlipRowDtos);
+
+        return accountingSlipDto;
+    }
+
+    public List<AccountingSlipDto> createAccountingSlips(Map<String, List<String>> accountingIdsByDate) {
+        List<AccountingSlipDto> accountingSlips = new ArrayList<>();
+
+        for (Map.Entry<String, List<String>> accountingsForDate : accountingIdsByDate.entrySet()) {
+            AccountingSlipDto accountingSlipDto = createAccountingSlipForDate(accountingsForDate);
+            accountingSlips.add(accountingSlipDto);
+        }
+        return accountingSlips;
+    }
+
+    public Map<String, List<String>> groupAccountingsByDate(List<Order> ordersToAccount) {
+        Map<String, List<String>> map = new HashMap<>();
+
+        List<String> orderIds = ordersToAccount.stream()
+                .map(Order::getOrderId)
+                .collect(Collectors.toList());
+
+        List<OrderAccounting> orderAccountings = orderAccountingService.getOrderAccountings(orderIds);
+
+        for (OrderAccounting orderAccounting : orderAccountings) {
+            String createdAt = orderAccounting.getCreatedAt();
+
+            if (LocalDate.parse(createdAt).isBefore(LocalDate.now())) {
+                List<String> accountingsForDate = map.get(createdAt);
+
+                if (accountingsForDate == null) {
+                    ArrayList<String> accountings = new ArrayList<>();
+                    accountings.add(orderAccounting.getOrderId());
+                    map.put(createdAt, accountings);
+                } else {
+                    accountingsForDate.add(orderAccounting.getOrderId());
+                }
+            }
+        }
+
+        return map;
+    }
+
+    private AccountingSlipDto createAccountingSlipForDate(Map.Entry<String, List<String>> accountingsForDate) {
+        List<OrderItemAccountingDto> itemAccountings = getOrderItemAccountingsForDate(accountingsForDate);
+
```

Comment:
english only, and if the functional approach is easier to write with simple logic in the dto classes then that's ok too